source/developer/interactive-messages.rst

```diff
@@ -0,0 +1,329 @@
+
+    .. code-block:: text
+
+      {
+      "user_id": "rd49ehbqyjytddasoownkuqrxe",
+      "post_id": "gqrnh3675jfxzftnjyjfe4udeh",
+      "channel_id": "j6j53p28k6urx15fpcgsr20psq",
+      "team_id": "5xxzt146eax4tul69409opqjlf",
+      "context": {
+        "repo": "mattermost/mattermost-server"
+        "pr": 1234,
+        "action": "merge",
+        "signature": "mycryptographicsignature"
+        }
+      }
+
+  It's also possible for integrations to do both of these things with a single token and use something like this as context:
+
+    .. code-block:: text
+
+      {
+      "user_id": "rd49ehbqyjytddasoownkuqrxe",
+      "post_id": "gqrnh3675jfxzftnjyjfe4udeh",
+      "channel_id": "j6j53p28k6urx15fpcgsr20psq",
+      "team_id": "5xxzt146eax4tul69409opqjlf",
+      "context": {
+        "action_id": "someunguessableactionid"
+        }
+      }
+
+  Then, when the integration receives the request, it can act based on the action id.
+
+Tips and Best Practices
+------------------------
+
+1. The external application may be written in any programming language. It needs to provide a URL which receives the request sent by your Mattermost server and responds with in the required JSON format.
+2. To get started, you can use this `sample plugin <https://github.com/matterpoll/matterpoll>`_ to add polling to Mattermost channels via a `/poll` slash command.
+
+Share Your Integration
+-----------------------
+
+If you've built an integration for Mattermost, please consider `sharing your work <https://www.mattermost.org/share-your-mattermost-projects/>`_ in our `app directory <https://about.mattermost.com/default-app-directory/>`_.
+
+The `app directory <https://about.mattermost.com/default-app-directory/>`_ lists open source integrations developed by the Mattermost community and are available for download, customization and deployment to your private cloud or on-prem infrastructure.
+
+Slack Compatibility
+--------------------
+
+Like Slack, actions are specified in an "actions" list within the message attachment. Moreover, your integrations can react with ephemeral messages or message updates similar to Slack.
+
+However, the schema for these objects is slightly different given Slack requires a Slack App and action URL to be pre-configured beforehand. Mattermost instead allows an integration to create an interactive message without pre-configuration.
+
+Troubleshooting
+--------------------
+
+Interactive messages don't show up for slash commands
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Make sure the `response type <https://docs.mattermost.com/developer/slash-commands.html#message-type>`_ of your slash command is set to ``in_channel``, not ``ephemeral``.
+
+Ephemeral messages do not have a state, and therefore do not support interactive message buttons at this time.
+
+Message buttons and menus do not trigger an action and return a 400 error
```

Comment:
This title doesn't render for me, I think it may need to use `^^^^` to match what was used earlier in the doc